doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1522,144 +1507,122 @@ Managing the local context
+   .. todo PR The only difference vs assert is where the goal is added?  If so, then
+      this entry should only refer to assert and mention the difference(s).
+      There are several similar tactics in this section.  Maybe we should add a
+      summary of their differences?  Perhaps you could briefly outline them?
 
-   This is an alternative syntax for :n:`eassert (@ident := @term)`
-   and :n:`epose proof @term as @ident`, following the model of
-   :n:`epose (@ident := @term)` but dropping the value of
-   :token:`ident`.
+   The first form adds a new hypothesis :n:`@ident : @type` to the context
+   and adds :n:`@type` as a new subgoal after the current goal (rather than
+   before it as :tacn:`assert` would).  If present, :n:`@ltac_expr3` must solve the
+   initial goal entirely after the extra assumption :n:`@type` is added and possibly
+   destructed; otherwise, the tactic will fail.
 
-.. tacv:: enough (@ident : @type)
-   :name: enough
+   .. todo PR: must succeed for all subgoals and if no, the tactic fails?
+      and applies to assert, too?
 
-   This adds a new hypothesis of name :token:`ident` asserting :token:`type` to the
-   goal the tactic :tacn:`enough` is applied to. A new subgoal stating :token:`type` is
-   inserted after the initial goal rather than before it as :tacn:`assert` would do.
+   In the second form, if :n:`@as_ipat` isn't specified, the tactic adds a new
+   hypothesis with a name chosen by Coq.  Otherwise, it transforms :n:`@type`
+   as specified by :n:`@as_ipat` and adds the resulting new hypotheses.
+   If the :n:`as @simple_intropattern` clause generates more than one subgoal,
+   :n:`@ltac_expr3`, if present, is applied to all of them.
 
-.. tacv:: enough @type
+   .. tacn:: eenough ( @ident : @type ) {? by @ltac_expr3 }
+             eenough @one_term {? @as_ipat } {? by @ltac_expr3 }
+      :name: eenough; _
 
-   This behaves like :n:`enough (@ident : @type)` with the name :token:`ident` of
-   the hypothesis generated by Coq.
+      Unlike :tacn:`enough`, the :n:`@type`\s and :n:`@one_term`\s in
+      :tacn:`eenough` may contain :term:`holes <hole>`, denoted by :n:`_`,
+      for which the tactic will create existential variables.  This lets you
+      avoid specifying the asserted statement completely before starting to
+      prove it.
 
-.. tacv:: enough @type as @simple_intropattern
+.. tacn:: cut @one_term
 
-   This behaves like :n:`enough @type` using :token:`simple_intropattern` to name or
-   destruct the new hypothesis.
+   Implements the non-dependent case of the :ref:`App <app_rule>` typing rule,
+   the Modus Ponens inference rule.  It is equivalent to
+   :n:`enough (@ident: @one_term). revert @ident.`
+   This tactic is generally considered obsolete but it is still widely
+   used in old scripts.
 
-.. tacv:: enough (@ident : @type) by @tactic
-          enough @type {? as @simple_intropattern } by @tactic
+.. tacn:: pose proof @term {? @as_ipat }
```

Comment:
It is strange that this form uses `@term` and not `@one_term` like `pose` (especially given the optional `@as_ipat` clause). I wonder why this is the case and whether restricting the grammar to only accept `@one_term` would create lots of breakage.